Hex Paper/DocumentsViewController.swift

```diff
@@ -0,0 +1,165 @@
+    func deleteItems(at indexPaths: [IndexPath])
+    func segueToEditDocumentScene()
+    func refreshDocumentData()
+}
+
+class DocumentsViewController: UICollectionViewController {
+
+    let borderColor = UIColor(colorLiteralRed: 0.220, green: 0.506, blue: 0.153, alpha: 1.0).cgColor
+    let borderWidth: CGFloat = 1.0
+    let selectedBorderWidth: CGFloat = 3.0
+    
+    var delegate: DocumentsSceneDelegate?
+    var editMode: Bool = false {
+        didSet {
+            if editMode {
+                navigationItem.setRightBarButtonItems([doneButton], animated: true)
+                navigationItem.setLeftBarButtonItems([trashButton], animated: true)
+                trashButton.isEnabled = false
+                collectionView?.allowsMultipleSelection = true
+            } else {
+                navigationItem.setRightBarButtonItems(initialRightBarButtonItems, animated: true)
+                navigationItem.setLeftBarButtonItems(initialLeftBarButtonItems, animated: true)
+                for indexPath in indexPathsForSelectedItems ?? [] {
+                    let cell = collectionView?.cellForItem(at: indexPath)
+                    cell?.layer.borderWidth = borderWidth
+                }
+                collectionView?.allowsMultipleSelection = false
+            }
+        }
+    }
+
+    fileprivate var trashButton: UIBarButtonItem!
+    fileprivate var doneButton: UIBarButtonItem!
+    private var initialRightBarButtonItems: [UIBarButtonItem]?
+    private var initialLeftBarButtonItems: [UIBarButtonItem]?
+
+    @IBOutlet weak var addButton: UIBarButtonItem!
+    
+    @IBAction func addButtonAction(_ sender: UIBarButtonItem) {
+        delegate?.documentsSceneAddButtonWasTapped(self)
+    }
+    
+    @IBAction func trashButtonAction(_ sender: UIBarButtonItem) {
+        trashButton.isEnabled = false
+        delegate?.documentsSceneTrashButtonWasTapped(self)
+    }
+    
+    @IBAction func doneButtonAction(_ sender: UIBarButtonItem) {
+        editMode = false
+    }
+
+    @IBAction func editButtonAction(_ sender: UIBarButtonItem) {
+        editMode = true
+    }
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        delegate = DocumentsController()
+        collectionView?.allowsSelection = true
+        initialRightBarButtonItems = navigationItem.rightBarButtonItems
+        initialLeftBarButtonItems = navigationItem.leftBarButtonItems
+        
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done,
+                                         target: self,
+                                         action: #selector(DocumentsViewController.doneButtonAction))
+        self.doneButton = doneButton
+        let trashButton = UIBarButtonItem(barButtonSystemItem: .trash,
+                                          target: self,
+                                          action: #selector(DocumentsViewController.trashButtonAction))
+        self.trashButton = trashButton
+    }
```

Comment:
If you put these buttons in a closure when you declare them as properties like:
```
fileprivate var trashButton: UIBarButtonItem = {
    let trashButton = UIBarButtonItem(barButtonSystemItem: .trash, target: self, action: #selector(DocumentsViewController.trashButtonAction))
    return trashButton
}()
```
Does it still work? (Honestly not sure) If so, then this init code could be moved up and `viewDidLoad` would only be for code that applies to that happening.